### ISSUE TITLE
Running wasmtime demo fails when built with cg_clif

### ISSUE THREAD

**Issue by vi** [bjorn3/rustc_codegen_cranelift#1230]
If built with `rustc_codegen_cranelif`'s Cargo wrapper, [wasmtime's "Hello world"](https://docs.wasmtime.dev/lang-rust.html#hello-world) (with `wasmtime = "0.36.0"`) outputs:

```
Error: compilation settings are not compatible with the native host

Caused by:
    compilation setting "has_sse3" is enabled but not available on the host
```

If I use usual Cargo, I get:

```
Answer: 42
```

cg_clif's Rust version: `cargo 1.62.0-nightly (a44758ac8 2022-05-04)`, repo version: 343c532f4f8bda436bf75250f9eb06d5aee245b4.
Usual Rust version: `cargo 1.63.0-nightly (a4c1cd0eb 2022-05-18)`.

**Comment by bjorn3** [comment]
Try using the main branch of wasmtime and I think disabling support for the simd wasm feature. Cg_clif fakes a cpuid output to disable most target features as the corresponding intrinsics aren't implemented yet. Wasmtime requires sse4.1 I believe when simd support is enabled. Also there was recently a fix for wasmtime requiring sse4.1 even when simd support is disabled. This fix isn't yet included in any released version of wasmtime

**Comment by bjorn3** [comment]
If you comment out https://github.com/bjorn3/rustc_codegen_cranelift/blob/c3a0431731e399b22670f83a7796649e50abd1b0/src/inline_asm.rs#L53-L132 it gets past this check but hits https://github.com/bjorn3/rustc_codegen_cranelift/issues/1381.

**Comment by bjorn3** [comment]
Fixed in 55380a5f. The entire wasmtime test suite passes now with the exception of the tests that test panics as cg_clif doesn't yet support unwinding on panics.
